worker/worker.go

```diff
@@ -150,43 +154,42 @@ func detectContent(name, path, imageFormat string, parent *database.Layer) (name
 	return
 }
 
-func detectNamespace(data map[string][]byte, parent *database.Layer) (namespace *database.Namespace, err error) {
-	namespace = detectors.DetectNamespace(data)
+func detectNamespaces(data map[string][]byte, parent *database.Layer) (namespaces []database.Namespace, err error) {
+	namespaces = detectors.DetectNamespaces(data)
+
+	// Inherit the non-detected namespace from its parent
+	if parent != nil {
+		mapNamespaces := make(map[string]database.Namespace)
+		for _, pn := range parent.Namespaces {
+			name := strings.Split(pn.Name, ":")
```

Comment:
So, should I add a `//TODO` to note it and add a version property for namespaces in v2 ?